src/db/add_package.rs

```diff
@@ -159,13 +158,12 @@ pub fn add_package_into_database(conn: &Connection,
         if let Some(versions_array) = versions.as_array_mut() {
             let mut found = false;
             for version in versions_array.clone() {
-                if &semver::Version::parse(version.as_string().unwrap()).unwrap() ==
-                   pkg.manifest().version() {
+                if version.as_string().unwrap() == metadata_pkg.version {
```

Comment:
hm so semver equality is not quite the same as string equality IIRC, so maybe this does need to parse?